### DISCUSSION TITLE
Is it possible to force a line feed (or carriage return) in a form

### DISCUSSION THREAD

**Discussion by kevinr432** [sqlpage/SQLPage#673] [General]
Is it possible to force a line feed (or carriage return) in a form before a particular form field.    I would like to have more control over how my fields appear on a form.  See attached.
![LineFeed](https://github.com/user-attachments/assets/6733c34c-46b9-45b1-bca8-a84e0715316c)

**Comment by lovasoa** [comment]
Hello, and welcome to SQLPage !

This is not possible with sqlpage natively.

You can either: 
 - enlarge the previous fields so that they take the entire line,
 -  or add custom css to your page using the shell component. In CSS, you can add a right margin to the Volume input field to force "AnnPrem" to wrap to the next line, for instance.
     - **`form.sql`** : 
       ```sql
       select 'number' as type, 'Volume' as name, 'volume-input' as id;
       ```
     - **`style.css`**
       ```css
       #volume-input { margin-right: 50% }
       ```

![image](https://github.com/user-attachments/assets/f436603d-9180-43cf-a491-2eba0cbf9506)

**Comment by kevinr432** [comment]
I have become fairly proficient already with SQL Page, but I don't know anything about style sheets.    This may sound like a stupid question, but where can I find the style sheet to edit it?    What I love about SQL Page is that it mostly insulates me from having to know HTML and CSS.

**Reply by lovasoa** [reply]
Yes, SQLPage allows you to build a full application without having to know HTML, CSS, or Javascript. But it requires you to make your application "fit" into the (growing) set of component offered by SQLPage. When you want to do something more custom, you still have to either wait for sqlpage to integrate it, or implement it yourself using traditional web languages.

You can add a spreadsheet to your application using the `css` attribute of the shell component:

```sql
select 'shell' as component, '/my_styles.css' as css;
```

https://sql.datapage.app/component.sql?component=shell#component
